pkg/controller/che/che_controller.go

```diff
@@ -1160,13 +1160,16 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 	desiredImagePullPolicy := util.GetValue(string(instance.Spec.Server.CheImagePullPolicy), deploy.DefaultPullPolicyFromDockerImage(cheImageAndTag))
 	effectiveImagePullPolicy := string(effectiveCheDeployment.Spec.Template.Spec.Containers[0].ImagePullPolicy)
 	desiredSelfSignedCert := instance.Spec.Server.SelfSignedCert
+	desiredCustomPublicCerts := instance.Spec.Server.ServerTrustStoreConfigMapName != ""
 	desiredGitSelfSignedCert := instance.Spec.Server.GitSelfSignedCert
 	effectiveSelfSignedCert := r.GetDeploymentEnvVarSource(effectiveCheDeployment, "CHE_SELF__SIGNED__CERT") != nil
+	effectiveCustomPublicCerts := r.GetDeploymentVolume(effectiveCheDeployment, "che-public-certs").ConfigMap != nil
 	effectiveGitSelfSignedCert := r.GetDeploymentEnvVarSource(effectiveCheDeployment, "CHE_GIT_SELF__SIGNED__CERT") != nil
 	if desiredMemRequest.Cmp(effectiveMemRequest) != 0 ||
 		desiredMemLimit.Cmp(effectiveMemLimit) != 0 ||
 		effectiveImagePullPolicy != desiredImagePullPolicy ||
 		effectiveSelfSignedCert != desiredSelfSignedCert ||
+		effectiveCustomPublicCerts != desiredCustomPublicCerts ||
```

Comment:
It seems that a change in the content of the config map will not trigger a redeploy of the impacted components (mainly the Che server).
You might want to also support this case by doing the same way as what is done for the `che` config map (with the current config map resource version being added as a metadata in the Che deployment itself).

@tolusha Do we want to support automatic rolling update of the Che server when the custom public certs config map content is changed ?

Reply:
yes. it make sense.

Reply:
The certificates are represented in files, so we create the certificates config-map with the help of  the `--from-file` key. It is impossible to add/edit a single property using files, user can just edit the whole content of the config map, but I don't think it will be the case.

Reply:
I have no idea how to check the certificates config-map changed states. In case of the `che config-map` there is a predefined list of values which is compared with the actual values, but the  certificates config map hasn't any predefined values. @tolusha @AndrienkoAleksandr @mmorhun @davidfestal WDYT?